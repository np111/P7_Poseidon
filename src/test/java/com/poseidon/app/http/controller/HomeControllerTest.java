package com.poseidon.app.http.controller;

import org.junit.jupiter.api.Test;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;

@WebMvcTest(HomeController.class)
class HomeControllerTest extends AbstractControllerTest {
    @Test
    void home() throws Exception {
        toMatchSnapshot(
                mockMvc.perform(get("/")).andReturn()
        );
    }

    @Test
    void adminHome() throws Exception {
        toMatchSnapshot(
                mockMvc.perform(get("/admin/home")).andReturn()
        );
    }
}