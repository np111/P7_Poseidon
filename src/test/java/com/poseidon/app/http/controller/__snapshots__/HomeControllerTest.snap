com.poseidon.app.http.controller.HomeControllerTest.adminHome=[
-
> GET /admin/home
< HTTP 302
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: http://localhost/login
""
]


com.poseidon.app.http.controller.HomeControllerTest.home=[
-
> GET /
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <title> Spring Boot </title>"
"  <link rel=\"stylesheet\" href=\"/css/styles.css\"/>"
"</head>"
"<body>"
"<h3> HOME PAGE </h3>"
"<p>"
"  <a href=\"/login\">Login</a> or create one <a href=\"/user/list\">User management</a>"
"</p>"
"</body>"
"</html>"
]